Build fails under go1.16

`go version`

```
go version go1.16rc1 linux/amd64
```

From go1.16 changelog:

> Build commands like go build and go test no longer modify go.mod and go.sum by default. Instead, they report an error if a module requirement or checksum needs to be added or updated (as if the -mod=readonly flag were used). Module requirements and sums may be adjusted with go mod tidy or go get.

`xcaddy build`
```
2021/02/05 10:29:01 [INFO] Temporary folder: /tmp/buildenv_2021-02-05-1029.946899145
2021/02/05 10:29:01 [INFO] Writing main module: /tmp/buildenv_2021-02-05-1029.946899145/main.go
2021/02/05 10:29:01 [INFO] Initializing Go module
2021/02/05 10:29:01 [INFO] exec (timeout=10s): /usr/bin/go mod init caddy 
go: creating new go.mod: module caddy
go: to add module requirements and sums:
	go mod tidy
2021/02/05 10:29:01 [INFO] Pinning versions
2021/02/05 10:29:01 [INFO] exec (timeout=0s): /usr/bin/go get -d -v github.com/caddyserver/caddy/v2 
go get: added github.com/caddyserver/caddy/v2 v2.3.0
2021/02/05 10:29:02 [INFO] Build environment ready
2021/02/05 10:29:02 [INFO] Building Caddy
2021/02/05 10:29:02 [INFO] exec (timeout=0s): /usr/bin/go build -o /home/dionysius/Документы/Исходники/go/gitlab.com/activecdn/caddy -ldflags -w -s -trimpath 
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:30:2: missing go.sum entry for module providing package github.com/Masterminds/sprig/v3 (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:31:2: missing go.sum entry for module providing package github.com/alecthomas/chroma/formatters/html (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/caddyconfig/httpcaddyfile/serveroptions.go:25:2: missing go.sum entry for module providing package github.com/dustin/go-humanize (imported by github.com/caddyserver/caddy/v2/modules/logging); to add:
	go get github.com/caddyserver/caddy/v2/modules/logging@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/acmeserver/acmeserver.go:29:2: missing go.sum entry for module providing package github.com/go-chi/chi (imported by github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:29:2: missing go.sum entry for module providing package github.com/google/cel-go/cel (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:30:2: missing go.sum entry for module providing package github.com/google/cel-go/checker/decls (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:31:2: missing go.sum entry for module providing package github.com/google/cel-go/common/types (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:32:2: missing go.sum entry for module providing package github.com/google/cel-go/common/types/ref (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:33:2: missing go.sum entry for module providing package github.com/google/cel-go/common/types/traits (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:34:2: missing go.sum entry for module providing package github.com/google/cel-go/ext (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:35:2: missing go.sum entry for module providing package github.com/google/cel-go/interpreter/functions (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/logging/encoders.go:25:2: missing go.sum entry for module providing package github.com/jsternberg/zap-logfmt (imported by github.com/caddyserver/caddy/v2/modules/logging); to add:
	go get github.com/caddyserver/caddy/v2/modules/logging@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/encode/gzip/gzip.go:25:2: missing go.sum entry for module providing package github.com/klauspost/compress/gzip (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/encode/gzip); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/encode/gzip@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/encode/zstd/zstd.go:21:2: missing go.sum entry for module providing package github.com/klauspost/compress/zstd (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/encode/zstd); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/encode/zstd@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddytls/values.go:23:2: missing go.sum entry for module providing package github.com/klauspost/cpuid/v2 (imported by github.com/caddyserver/caddy/v2/modules/caddytls); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddytls@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/app.go:28:2: missing go.sum entry for module providing package github.com/lucas-clemente/quic-go/http3 (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/frontmatter.go:9:2: missing go.sum entry for module providing package github.com/naoina/toml (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/acmeserver/acmeserver.go:30:2: missing go.sum entry for module providing package github.com/smallstep/certificates/acme (imported by github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/acmeserver/acmeserver.go:31:2: missing go.sum entry for module providing package github.com/smallstep/certificates/acme/api (imported by github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/ca.go:28:2: missing go.sum entry for module providing package github.com/smallstep/certificates/authority (imported by github.com/caddyserver/caddy/v2/modules/caddypki); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddytls/internalissuer.go:29:2: missing go.sum entry for module providing package github.com/smallstep/certificates/authority/provisioner (imported by github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/ca.go:29:2: missing go.sum entry for module providing package github.com/smallstep/certificates/db (imported by github.com/caddyserver/caddy/v2/modules/caddypki); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/certificates.go:21:2: missing go.sum entry for module providing package github.com/smallstep/cli/crypto/x509util (imported by github.com/caddyserver/caddy/v2/modules/caddypki); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/acmeserver/acmeserver.go:35:2: missing go.sum entry for module providing package github.com/smallstep/nosql (imported by github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki/acmeserver@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddypki/ca.go:30:2: missing go.sum entry for module providing package github.com/smallstep/truststore (imported by github.com/caddyserver/caddy/v2/modules/caddypki); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddypki@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:34:2: missing go.sum entry for module providing package github.com/yuin/goldmark (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:35:2: missing go.sum entry for module providing package github.com/yuin/goldmark-highlighting (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:36:2: missing go.sum entry for module providing package github.com/yuin/goldmark/extension (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:37:2: missing go.sum entry for module providing package github.com/yuin/goldmark/parser (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/tplcontext.go:38:2: missing go.sum entry for module providing package github.com/yuin/goldmark/renderer/html (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/celmatcher.go:36:2: missing go.sum entry for module providing package google.golang.org/genproto/googleapis/api/expr/v1alpha1 (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/logging/filewriter.go:28:2: missing go.sum entry for module providing package gopkg.in/natefinch/lumberjack.v2 (imported by github.com/caddyserver/caddy/v2/modules/logging); to add:
	go get github.com/caddyserver/caddy/v2/modules/logging@v2.3.0
/home/dionysius/go/pkg/mod/github.com/caddyserver/caddy/v2@v2.3.0/modules/caddyhttp/templates/frontmatter.go:10:2: missing go.sum entry for module providing package gopkg.in/yaml.v2 (imported by github.com/caddyserver/caddy/v2/modules/caddyhttp/templates); to add:
	go get github.com/caddyserver/caddy/v2/modules/caddyhttp/templates@v2.3.0
2021/02/05 10:29:02 [INFO] Cleaning up temporary folder: /tmp/buildenv_2021-02-05-1029.946899145
2021/02/05 10:29:02 [FATAL] exit status 1
```

--- a/builder.go
+++ b/builder.go
@@ -94,7 +94,9 @@ func (b Builder) Build(ctx context.Context, outputFile string) error {
 
 	// tidy the module to ensure go.mod and go.sum are consistent with the module prereq
 	tidyCmd := buildEnv.newCommand("go", "mod", "tidy")
-	buildEnv.runCommand(ctx, tidyCmd, b.TimeoutGet)
+	if err := buildEnv.runCommand(ctx, tidyCmd, b.TimeoutGet); err != nil {
+		return err
+	}
 
 	// compile
 	cmd := buildEnv.newCommand("go", "build",

--- a/builder.go
+++ b/builder.go
@@ -92,6 +92,10 @@ func (b Builder) Build(ctx context.Context, outputFile string) error {
 
 	log.Println("[INFO] Building Caddy")
 
+	// tidy the module to ensure go.mod and go.sum are consistent with the module prereq
+	tidyCmd := buildEnv.newCommand("go", "mod", "tidy")
+	buildEnv.runCommand(ctx, tidyCmd, b.TimeoutGet)
+
 	// compile
 	cmd := buildEnv.newCommand("go", "build",
 		"-o", absOutputFile,

